test/buildAndRun.test.ts

```diff
@@ -0,0 +1,120 @@
+            let relativePath = entry.entryName.slice(sourceFolderInZip.length);
+            if (!relativePath) {
+                // root folder
+                continue;
+            }
+
+            let outPath = path.join(outputFolder, relativePath);
+            if (entry.isDirectory) {
+                //console.log(`Folder: ${entry.entryName}`);
+                await fse.mkdirs(outPath)
+            } else {
+                //console.log(`File: ${entry.entryName}`);
+                let data: Buffer = entry.getData();
+                await fse.writeFile(outPath, data);
+            }
+        }
+    }
+}
+
+suite("Build Image", function (this: Suite): void {
+    this.timeout(Math.max(60 * 1000, this.timeout()));
+
+    const outputChannel: vscode.OutputChannel = vscode.window.createOutputChannel('Docker extension tests');
+    ext.outputChannel = outputChannel;
+
+    async function testConfigureAndBuildImage(
+        platform: Platform,
+        configureInputs: (string | undefined)[],
+        buildInputs: (string | undefined)[]
+    ): Promise<void> {
+        // Set up simulated user input
+        configureInputs.unshift(platform);
+        ext.ui = new TestUserInput(configureInputs);
+        let testTerminalProvider = new TestTerminalProvider();
+        ext.terminalProvider = testTerminalProvider;
+
+        await configure(testRootFolder);
+        assert.equal(configureInputs.length, 0, 'Not all inputs were used for configure docker files');
+
+        // Build image
+        ext.ui = new TestUserInput(buildInputs);
+        let dockerFile = Uri.file(path.join(testRootFolder, 'Dockerfile'));
+        await commands.executeCommand('vscode-docker.image.build', dockerFile);
+        assert.equal(configureInputs.length, 0, 'Not all inputs were used for Build Image');
+
+        let { outputText, errorText } = await testTerminalProvider.currentTerminal.exit();
+
+        console.log("=== OUTPUT BEGIN ================================");
+        console.log(outputText ? outputText : '(NONE)');
+        console.log("=== OUTPUT END ==================================");
+
+        console.log("=== ERROR OUTPUT BEGIN ================================");
+        console.log(errorText ? errorText : '(NONE)');
+        console.log("=== ERROR OUTPUT END ==================================");
+
+        assert.equal(errorText, '', 'Expected no errors from Build Image');
+        assertEx.assertContains(outputText, 'Successfully built');
+        assertEx.assertContains(outputText, 'Successfully tagged')
+    }
+
+    function testInEmptyFolder(name: string, func: () => Promise<void>): void {
+        test(name, async () => {
+            // Delete everything in the root testing folder
+            assert(path.basename(testRootFolder) === constants.testOutputName, "Trying to delete wrong folder");;
+            await fse.emptyDir(testRootFolder);
+            await func();
+        });
+    }
+
+    // Node.js
```

Comment:
All languages have tests. When Ruby was recently added I asked them to add the test.

Reply:
Starting with one first (specifically because there was a bug about it).  Not sure if I'll get to them all, but added this:

    // NEEDED TESTS:
    // 'Java'
    // '.NET Core Console'
    // 'ASP.NET Core'
    // 'Node.js'
    // 'Python'
    // 'Ruby'